bin/metrics/zenossStatsView.py

```diff
@@ -84,21 +86,29 @@ def get_zopes(self, first_time=False):
             self.zopes = {}
 
         # Check mtime of /opt/zenoss/zproxy/conf/zope-upstreams.conf
-        # if it's newer than, say, now - self.interval, reread it.
+        # if it's newer than it was modified last time, reread it.
         zope_upstream_file = self.ZPROXY_CONF_DIR + 'zope-upstreams.conf'
         zenapi_upstream_file = self.ZPROXY_CONF_DIR + 'apizopes-upstreams.conf'
         zenreports_upstream_file = self.ZPROXY_CONF_DIR + 'zopereports-upstreams.conf'
         zauth_upstream_file = self.ZPROXY_CONF_DIR + 'zauth-upstreams.conf'
 
+        def read_upstream_file(upstream_file):
+            with open(upstream_file, 'r') as inf:
+                zopes = inf.readlines()
+                zopes = [line.rstrip('\n;') for line in zopes]
+                zopes = [line.split(' ')[-1] for line in zopes]
+            return zopes
+
         def check_upstream_util(upstream_file):
             upstream_modified = os.path.getmtime(upstream_file)
-            now = time.time()
             zopes = []
-            if first_time or upstream_modified > (now - self.interval):
-                with open(upstream_file, 'r') as inf:
-                    zopes = inf.readlines()
-                    zopes = [line.rstrip('\n;') for line in zopes]
-                    zopes = [line.split(' ')[-1] for line in zopes]
+            if first_time:
+                now = time.time()
+                zopes = read_upstream_file(upstream_file)
+                self.last_changed[upstream_file] = now
```

Comment:
while this will work, shouldn't we store upstream_modified instead, since we have it?